starport/templates/typed/stargate/component/x/{{moduleName}}/keeper/{{typeName}}_test.go.plush

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/assert"
+    "<%= ModulePath %>/x/<%= ModuleName %>/types"
+)
+
+func createN<%= title(TypeName) %>(keeper *Keeper, ctx sdk.Context, n int) []types.<%= title(TypeName) %> {
+	items := make([]types.<%= title(TypeName) %>, n)
+	for i := range items {
+		items[i].Creator = "any"
+		items[i].Id = keeper.Append<%= title(TypeName) %>(ctx, items[i])
+	}
+	return items
+}
+
+func Test<%= title(TypeName) %>Get(t *testing.T) {
+	keeper, ctx := setupKeeper(t)
+	items := createN<%= title(TypeName) %>(keeper, ctx, 10)
+	for _, item := range items {
+		assert.Equal(t, item, keeper.Get<%= title(TypeName) %>(ctx, item.Id))
+	}
+}
+
+func Test<%= title(TypeName) %>Exist(t *testing.T) {
+	keeper, ctx := setupKeeper(t)
+	items := createN<%= title(TypeName) %>(keeper, ctx, 10)
+	for _, item := range items {
+		assert.True(t, keeper.Has<%= title(TypeName) %>(ctx, item.Id))
+	}
+}
+
+func Test<%= title(TypeName) %>Remove(t *testing.T) {
+	keeper, ctx := setupKeeper(t)
+	items := createN<%= title(TypeName) %>(keeper, ctx, 10)
+	for _, item := range items {
+		keeper.Remove<%= title(TypeName) %>(ctx, item.Id)
+		assert.False(t, keeper.Has<%= title(TypeName) %>(ctx, item.Id))
+	}
+}
+
+func Test<%= title(TypeName) %>GetAll(t *testing.T) {
+	keeper, ctx := setupKeeper(t)
+	items := createN<%= title(TypeName) %>(keeper, ctx, 10)
+	assert.Equal(t, items, keeper.GetAll<%= title(TypeName) %>(ctx))
+}
+
+func Test<%= title(TypeName) %>Count(t *testing.T) {
+	keeper, ctx := setupKeeper(t)
+	items := createN<%= title(TypeName) %>(keeper, ctx, 10)
+	count := uint64(len(items))
+	assert.Equal(t, count, keeper.Get<%= title(TypeName) %>Count(ctx))
```

Comment:
```suggestion
	require.Equal(t, count, keeper.Get<%= title(TypeName) %>Count(ctx))
```

assert only returns a bool but require actually fails the test if assert returns false. can we make the smae change in similar places as well?

Reply:
the core difference is that require fails the test immediately, and assert fails the test only after all results.
in practice require should be used only when you know that if you don't terminate the test your program will crash with cryptic traceback

Reply:
or maybe not necessarily cryptic, but when you suspect that future errors will be affected by the current error

Reply:
Yes, I always prefer failing immediately to keep following tests safe but assert is also okay.

Reply:
i changed my mind at some point when i had to change a line during debugging to `assert` in order to see if the test was failing for another item in the collection. not in this case obviously, this one is rather obvious